swarm/network/simulation/kademlia_test.go

```diff
@@ -63,4 +85,47 @@ func TestWaitTillHealthy(t *testing.T) {
 			t.Fatal(err)
 		}
 	}
+
+	// now create a snapshot of this network
+	snap, err := sim.Net.Snapshot()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// create a control simulation
+	controlSim := New(simServiceMap)
+	defer controlSim.Close()
+
+	// load the snapshot into this control simulation
+	err = controlSim.Net.Load(snap)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// for each node...
+	nodeIDs := controlSim.UpNodeIDs()
+	for _, node := range nodeIDs {
+		// ...get its kademlia
+		item, ok := sim.NodeItem(node, BucketKeyKademlia)
+		if !ok {
+			t.Fatal("No kademlia bucket item")
+		}
+		kad := item.(*network.Kademlia)
+		// get its base address
+		kid := common.Bytes2Hex(kad.BaseAddr())
+		// build a PeerPot
+		addrs := [][]byte{kad.BaseAddr()}
+		kad.EachAddr(nil, 255, func(addr *network.BzzAddr, po int) bool {
```

Comment:
Curious: If you run this with `256` instead of `255` does it return "self"?

Reply:
Not sure, but in this case we **don't want** `self`, right?

Reply:
Well you seem to want to ;)

```
addrs := [][]byte{kad.BaseAddr()}
```

Reply:
> Curious: If you run this with `256` instead of `255` does it return "self"?

yes..  but in fact,  you need to create `pp` only once